Document store `update_embeddings` fails with DPR

**Describe the bug**
If you train a `DensePassageRetriever` with `num_hard_negatives` > 0, then doing the following:

```python
retriever = DensePassageRetriever(
    document_store=InMemoryDocumentStore(),
    query_embedding_model=query_model,
    passage_embedding_model=passage_model,
    max_seq_len_query=128,
    max_seq_len_passage=256,
)

retriever.train(
    data_dir=DATA_PATH,
    train_filename=train_filename,
    dev_filename=val_filename,
    test_filename=val_filename,
    n_epochs=4,
    batch_size=16,
    grad_acc_steps=8,
    save_dir=dpr_model_dir,
    evaluate_every=3000,
    embed_title=False,
    num_positives=1,
    num_hard_negatives=5,

document_store = FAISSDocumentStore(similarity="dot_product")

docs = [] # insert docs here

document_store.write_documents(docs)

document_store.update_embeddings(retriever)
```

will trigger this assert: https://github.com/deepset-ai/haystack/blob/c5138655660b047d37e6628115af1315ca5c782e/haystack/document_store/faiss.py#L261


**Error message**
```python
In [55]: document_store.update_embeddings(retriever)
07/26/2021 15:39:22 - INFO - haystack.document_store.faiss -   Updating embeddings for 100467 docs...
Updating Embedding:   0%|                                                                                                                                                                                                              | 0/100467 [05:09<?, ? docs/s]
---------------------------------------------------------------------------                                                                                                                                                                                          
AssertionError                            Traceback (most recent call last)
<ipython-input-55-a042bbe71db2> in <module>
----> 1 document_store.update_embeddings(retriever)

~/miniconda3/envs/pytorch/lib/python3.7/site-packages/haystack/document_store/faiss.py in update_embeddings(self, retriever, index, update_existing_embeddings, filters, batch_size)
    259             for document_batch in batched_documents:
    260                 embeddings = retriever.embed_passages(document_batch)  # type: ignore
--> 261                 assert len(document_batch) == len(embeddings)
    262 
    263                 embeddings_to_index = np.array(embeddings, dtype="float32")

AssertionError: 
```

--- a/haystack/retriever/dense.py
+++ b/haystack/retriever/dense.py
@@ -269,6 +269,12 @@ class DensePassageRetriever(BaseRetriever):
         :param docs: List of Document objects used to represent documents / passages in a standardized way within Haystack.
         :return: Embeddings of documents / passages shape (batch_size, embedding_dim)
         """
+
+        if self.processor.num_hard_negatives != 0:
+            logger.warning(f"'num_hard_negatives' is set to {self.processor.num_hard_negatives}, but inference does "
+                           f"not require any hard negatives. Setting num_hard_negatives to 0.")
+            self.processor.num_hard_negatives = 0
+
         passages = [{'passages': [{
             "title": d.meta["name"] if d.meta and "name" in d.meta else "",
             "text": d.text,

